src/lib/concurrency/transaction_context.cpp

```diff
@@ -10,11 +10,13 @@
 
 namespace opossum {
 
-TransactionContext::TransactionContext(const TransactionID transaction_id, const CommitID snapshot_commit_id)
+TransactionContext::TransactionContext(const TransactionID transaction_id, const CommitID snapshot_commit_id,
+                                       bool is_auto_commit)
```

Comment:
Same thing as in the previous comment (hoping Github keeps the order): bool parameters make it hard to understand the function call. Same for `rollback` and `_mark_as_rolled_back`